### PR TITLE
EDC Extension - Authenticate on Data-Management API with rotating token

### PR DESCRIPTION
**Pull request by dileep-ramachandrarao**
As described in this [Jira task](https://jira.catena-x.net/browse/A1IDSC-211), this extension makes use of opensource [Java-security client library](https://github.com/SAP/cloud-security-xsuaa-integration/tree/main/java-security) to verify the incoming token on data management API.

### CODE REVIEW

**Comment by DominikPinsel** [comment]
Thanks for your contribution. We have taken a look at the code and would like to provide you with some feedback.


1. The Copyright claims are not updated yet. As the extension is provided by you, feel free to add them for your company :)
2. I'm not sure whether the majority of EDC users intends to use this extension in the future instead of the TokenBasedAuthenticationService. Would it be possible to remove your extension from the InMemory Control Plane for now? Unfortunatelly it's not possible to have multiple extensions implementing the same interface in the EDC (in this case AuthenticationService). So we have to go with a single option for now.
3. For code coverage currently it's necessary to add your extension to [this](https://github.com/catenax-ng/product-edc/blob/develop/edc-tests/pom.xml) pom file
4. It would be great if your code could pass the Catena-X SonarQube quality gate, before we merge your PR.

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Mercedes-Benz Tech Innovation GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

**Comment by dileep-ramachandrarao** [comment]
Hi @DominikPinsel,
I have updated the code as per your suggestions above.
However, there is an error in the CI workflow: https://github.com/catenax-ng/product-edc/actions/runs/3016976315
Any idea how it could be resolved?

**Comment by dileep-ramachandrarao** [comment]
Hi @DominikPinsel,
Could you please approve the awaiting workflows, so that I could evaluate if all the checks are successful or not.
Thanks in advance

**Comment by dileep-ramachandrarao** [comment]
@DominikPinsel please review and suggest next steps.
Thanks in advance

File: edc-extensions/xsuaa-authenticator/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+This extension makes use of opensource [Java-security client library](https://github.com/SAP/cloud-security-xsuaa-integration/tree/main/java-security) to verify the incoming token on data management API.
+
+This extension could also serve as an implementation template for other OAuth 2.0 services.
+
+**Please note** The service configuration needs to be loaded as described [here](https://github.com/SAP/cloud-security-xsuaa-integration/tree/main/java-security#setup-step-1-load-the-service-configurations). 
+
+For local testing, please provide the values for the following environment variables: `XSUAA_APP_ID`, `XSUAA_UAA_DOMAIN`, `XSUAA_AUTH_URL`, `XSUAA_CLIENT_ID`, `XSUAA_CLIENT_SECRET`
+
+In Kubernetes environment, please configure as described [here](https://github.com/SAP/cloud-security-xsuaa-integration/tree/main/java-security#mega-service-configuration-in-kuberneteskyma-environment), so that the client library could load the service configuration values.
```
Comment (florianrusch-zf) [inline edc-extensions/xsuaa-authenticator/README.md:13]: Can we make this maybe available within this README? The SAP repo is out of our hands and if someone decides in the future to remove it, the documentation would be also gone for us, which we should/must avoid!

**Comment by dileep-ramachandrarao** [comment]
@florianrusch-zf, could you please approve the awaiting workflows. 
Thanks in advance

**Comment by DominikPinsel** [comment]
> @florianrusch-zf, could you please approve the awaiting workflows. Thanks in advance

Sorry I was on vaccation. Workflow is approved :)

**Comment by dileep-ramachandrarao** [comment]
@DominikPinsel @florianrusch-zf , Could you please review and merge OR suggest if any changes are needed.

**Comment by lholthof** [comment]
@DominikPinsel, @florianrusch-zf, I am waiting for this PR to be merged, can you please support us to do so. We do not have write access to the repository and hence we cannot merge.

**Comment by DominikPinsel** [comment]
@dileep-ramachandrarao @lholthof sorry for the delay, thank you very much for the contribution and the next time our review will be faster :-)

<sub>Dominik Pinsel <dominik.pinsel@mercedes-benz.com>, Mercedes-Benz Tech Innovation GmbH, [legal info/Impressum](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md) </sub>

**Pull request merged**